### PR TITLE
refactor(orc8r): Segregation of state.proto & its servicers

### PR DESCRIPTION
**Pull request by shivesh-wavelabs**
Signed-off-by: shivesh <shivesh.ojha@wavelabs.ai>

## Summary
This PR deals with the segregation of state.proto servicers internal gRPC servicers implementation. GetState as internal services by separating the state proto into 2 new files made changes in state servicer as required. These changes are part of the ongoing work which details the distinction between Orc8r-internal gRPC endpoints vs. external, gateway-oriented gRPC endpoints.

Verified existing UT cases.

This PR was done as part of discussion by @hcgatewood [RemoveGatewayAccess toOrc8r-InternalEndpoints](https://drive.google.com/file/d/1NO6Qd6rU80xPh0Sb0mKWlSfDt0JbeVIT/view)

## Tests
All UT Passed

### CODE REVIEW

File: orc8r/cloud/go/services/state/protos/internal_state.proto [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/*
```
Comment (sabithaatla) [inline orc8r/cloud/go/services/state/protos/internal_state.proto]: Proto name should be state.proto

File: orc8r/cloud/go/services/state/state/main.go [hunk 2]
```diff
@@ -70,14 +72,17 @@ func main() {
...
+
+	stateExternalServicer := newExternalStateServicer(store)
+	protos.RegisterStateServiceServer(srv.GrpcServer, stateExternalServicer)
 
 	singletonReindex := srv.Config.MustGetBool(state_config.EnableSingletonReindex)
 	if !singletonReindex {
 		glog.Info("Running reindexer")
 		indexerManagerServer := newIndexerManagerServicer(srv.Config, db, store)
-		indexer_protos.RegisterIndexerManagerServer(srv.GrpcServer, indexerManagerServer)
+		internal_protos.RegisterIndexerManagerServer(srv.GrpcServer, indexerManagerServer)
```
Comment (sabithaatla) [inline orc8r/cloud/go/services/state/state/main.go:85]: why is this change required? i can see Indexer.proto is not moved

File: orc8r/cloud/go/services/state/state/main.go [hunk 3]
```diff
@@ -70,14 +72,17 @@ func main() {
 		glog.Fatalf("Error initializing state database: %v", err)
 	}
 
-	stateServicer := newStateServicer(store)
-	protos.RegisterStateServiceServer(srv.GrpcServer, stateServicer)
+	stateInternalServicer := newInternalStateServicer(store)
```
Comment (sabithaatla) [inline orc8r/cloud/go/services/state/state/main.go:75]: where is this defined?

File: orc8r/cloud/go/services/state/state/main.go [hunk 4]
```diff
@@ -70,14 +72,17 @@ func main() {
...
+	internal_protos.RegisterStateServiceServer(srv.GrpcServer, stateInternalServicer)
+
+	stateExternalServicer := newExternalStateServicer(store)
```
Comment (sabithaatla) [inline orc8r/cloud/go/services/state/state/main.go:78]: where is this defined?

File: orc8r/cloud/go/services/state/protos/internal_state.proto (hunk 1 above)
Comment (shivesh-wavelabs) [reply orc8r/cloud/go/services/state/protos/internal_state.proto]: Modified it to state.proto

File: orc8r/cloud/go/services/state/servicers/southbound/servicer.go [hunk 5]
```diff
@@ -198,37 +189,37 @@ func (srv *stateServicer) SyncStates(ctx context.Context, req *protos.SyncStates
...
-		return nil, internalErr(err, "GetStates (get) blobstore commit transaction")
+		return nil, common.InternalErr(err, "GetStates (get) blobstore commit transaction")
 	}
 
-	return &protos.GetStatesResponse{States: blobsToStates(blobs)}, nil
+	return &internal_protos.GetStatesResponse{States: blobsToStates(blobs)}, nil
 }
 
-func (srv *stateServicer) searchStates(_ context.Context, req *protos.GetStatesRequest) (*protos.GetStatesResponse, error) {
+func (srv *stateServicer) searchStates(_ context.Context, req *internal_protos.GetStatesRequest) (*internal_protos.GetStatesResponse, error) {
```
Comment (sabithaatla) [inline orc8r/cloud/go/services/state/servicers/southbound/servicer.go]: where is this called ?

File: orc8r/cloud/go/services/state/servicers/southbound/validate.go [hunk 6]
```diff
@@ -20,13 +20,6 @@ import (
 	"magma/orc8r/lib/go/protos"
 )
```
Comment (sabithaatla) [inline orc8r/cloud/go/services/state/servicers/southbound/validate.go:21]: Export the same function. why do we need to delete it?

File: orc8r/cloud/go/services/state/servicers/protected/servicer.go [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+	err = store.Commit()
+	if err != nil {
+		return nil, common.InternalErr(err, "GetStates (search) blobstore commit transaction")
+	}
+
+	return &protos.GetStatesResponse{States: blobsToStates(searchResults[req.NetworkID])}, nil
+}
+
+func isStateSynced(deviceIdToStates map[string][]*external_protos.State, reqIdAndVersion *external_protos.IDAndVersion) (bool, uint64) {
```
Comment (sabithaatla) [inline orc8r/cloud/go/services/state/servicers/protected/servicer.go]: where is this function called?

**Pull request closed without merging**